src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRunCodegenNodeFactory.cs

```diff
@@ -425,16 +425,20 @@ public IEnumerable<MethodWithGCInfo> EnumerateCompiledMethods(EcmaModule moduleT
             foreach (IMethodNode methodNode in MetadataManager.GetCompiledMethods(moduleToEnumerate, methodCategory))
             {
                 MethodDesc method = methodNode.Method;
-                Debug.Assert(methodNode == MethodEntrypoint(method));
                 MethodWithGCInfo methodCodeNode = methodNode as MethodWithGCInfo;
-                if (methodCodeNode == null && methodNode is LocalMethodImport localMethodImport)
```

Comment:
OK, so does this mean that the pre-existing logic for looking up methods with GC info in Local/PrecodeMethodImport nodes was basically superfluous so that you changed it into mere debug-specific diagnostic?

Reply:
Yes. Well, after my set of changes it became superfluous. It also turns out the fix to add more sorting reduced performance a fair bit as well. I've put some effort into writing a simple parallel merge sort algorithm. It appears to bring the perf back, and then some. In general the sorting needs in crossgen2 have an extremely expensive comparison routine, so changing to use classical merge sort yields performance improvements over Array.Sort, and parallelizing it is even better.